app/testOnly/controllers/SessionStorageServiceController.scala

```diff
@@ -48,52 +49,79 @@ class SessionStorageServiceController @Inject()(implicit val ec: ExecutionContex
-    hc.sessionId match {
-      case Some(sessionId: SessionId) => post(isAgent = isAgent, sessionId) flatMap {
-        case Left(ex) =>
-          Logger("application").error(s"${if (isAgent) "Agent" else "Individual"} - POST user data to income-tax-session-data unsuccessful: - ${ex.getMessage} - ${ex.getCause} - ")
-          Future.successful(handleError(isAgent))
-        case Right(id: String) =>
-          handlePostSuccess(id, isAgent)
-      }
-      case None =>
-        Logger("application").error(s"${if (isAgent) "Agent" else "Individual"}" +
-          s" - HeaderCarrier contained no sessionId!")
-        Future.successful(handleError(isAgent))
+    post() flatMap {
+      case Left(ex) =>
+        Logger("application")
+          .error(s"${if (isAgent) "Agent" else "Individual"} - POST user data to income-tax-session-data unsuccessful: - ${ex.getMessage} - ${ex.getCause} - ")
+        Future.successful(InternalServerError(s"There was an error. message: ${ex.getMessage}, cause: ${ex.getCause}"))
+      case Right(res: SessionDataPostSuccess) =>
+        handlePostSuccess(res, isAgent)
     }
   }.recover {
     case ex: Throwable =>
-      Logger("application").error(s"${if (isAgent) "Agent" else "Individual"} - Error on income-tax-session-data service test only page, status: - ${ex.getMessage} - ${ex.getCause} - ")
+      Logger("application").error(s"${if (isAgent) "Agent" else "Individual"}" +
+        s" - Error on income-tax-session-data service test only page, status: - ${ex.getMessage} - ${ex.getCause} - ")
       handleError(isAgent)
   }
 
-  private def handlePostSuccess(id: String, isAgent: Boolean)(implicit hc: HeaderCarrier, request: Request[_]): Future[Result] = {
-    sessionDataService.getSessionData(id) map {
+  private def handlePostSuccess(res: SessionDataPostSuccess, isAgent: Boolean)(implicit hc: HeaderCarrier): Future[Result] = {
+    sessionDataService.getSessionData(res.mtditid) map {
       case Left(ex) =>
         Logger("application").error(s"${if (isAgent) "Agent" else "Individual"}" +
           s" - GET user data request to income-tax-session-data unsuccessful: - ${ex.getMessage} - ${ex.getCause} - ")
         InternalServerError("Internal server error. There was an unexpected error fetching this data from income-tax-session-data service")
-      case Right(model: SessionDataModel) =>
+      case Right(model: SessionDataRetrieval) =>
         Ok(
-          s"User model:          ${model.toString}\n" +
-            s"session id:          ${model.sessionID}\n" +
-            s"mtditid:             ${model.mtditid}\n" +
-            s"nino:                ${model.nino}\n" +
-            s"saUtr:               ${model.saUtr}\n" +
-            s"client first name:   ${model.clientFirstName}\n" +
-            s"client last name:    ${model.clientLastName}\n" +
-            s"user type:           ${model.userType}\n")
+          s"Status:              ${res.status}\n" +
```

Comment:
Not sure how, but worth to consider showing response status code after POST to see if we are getting the right code etc